.github/workflows/doctests.yml

```diff
@@ -0,0 +1,80 @@
+name: Doctests
+
+on:
+  push:
+    branches:
+      - doctest*
```

Comment:
Don't we want to the doc tests to run on related changes on PRs too?

Reply:
On a second thought, I think it's fine to not run it on PRs since our doc tests will be much more computationally expensive. But I think I didn't see anything code that will prepare the separate "doctest-*" branches.